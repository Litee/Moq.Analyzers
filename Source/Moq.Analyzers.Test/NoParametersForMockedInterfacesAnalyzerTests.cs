using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.CodeAnalysis.Diagnostics;
using System.IO;
using TestHelper;
using Xunit;

namespace Moq.Analyzers.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class NoParametersForMockedInterfacesAnalyzerTests : DiagnosticVerifier
    {

        [Fact]
        public void ShouldFailIfMockedInterfaceHasParameters()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/MockInterfaceWithParameters.cs")));
        }

        [Fact]
        public void ShouldPassIfCustomMockClassIsUsed()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/MockInterfaceWithParametersCustomMockFile.cs")));
        }

        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new NoParametersForMockedInterfacesAnalyzer();
        }
    }
}