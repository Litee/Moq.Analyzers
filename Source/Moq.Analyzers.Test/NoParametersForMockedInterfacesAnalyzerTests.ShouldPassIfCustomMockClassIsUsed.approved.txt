Diagnostic 1
	Id: Moq1001
	Location: SourceFile(Test0.cs[613..622))
	Code: (1, true)
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 2
	Id: Moq1001
	Location: SourceFile(Test0.cs[715..720))
	Code: ("2")
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 3
	Id: Moq1001
	Location: SourceFile(Test0.cs[771..802))
	Code: (Moq.MockBehavior.Default, "3")
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 4
	Id: Moq1001
	Location: SourceFile(Test0.cs[895..924))
	Code: (MockBehavior.Loose, 4, true)
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 5
	Id: Moq1001
	Location: SourceFile(Test0.cs[975..997))
	Code: (MockBehavior.Default)
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 6
	Id: Moq1001
	Location: SourceFile(Test0.cs[1090..1112))
	Code: (MockBehavior.Default)
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

